src/main/java/cubicchunks/server/CubeProviderServer.java

```diff
@@ -80,6 +80,7 @@
     @Nonnull private ICubeIO cubeIO;
 
     // TODO: Use a better hash map!
+    @Nonnull private XYZMap<Cube> watchedCubeMap = new XYZMap<>(0.7f, 8000);
```

Comment:
IIRC PlayerCubeMap already contains this information. Why make another map for it?

Reply:
So you suggest to get cubes from PlayerCubeMap to do cube ticks?

Reply:
Cube watchers specifically. Or actually have `getCubeIterator` method in there, a cubic chunks equivalent of `getChunkIterator`.